pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble_suite_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assemble
+
+import (
+	"io/ioutil"
+	"testing"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	"github.com/oam-dev/kubevela/pkg/oam"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestAssemble(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Assemble Suite")
+}
+
+var _ = Describe("Test Assemble Options", func() {
+	It("test assemble", func() {
+		var (
+			compName  = "test-comp"
+			namespace = "default"
+		)
+
+		appRev := &v1beta1.ApplicationRevision{}
+		b, err := ioutil.ReadFile("./testdata/apprevision.yaml")
+		/* appRevision test data is generated based on below application
+		apiVersion: core.oam.dev/v1beta1
+		kind: Application
+		metadata:
+		  name: test-assemble
+		spec:
+		  components:
+		    - name: test-comp
+		      type: webservice
+		      properties:
+		        image: crccheck/hello-world
+		        port: 8000
+		      traits:
+		        - type: ingress
+		          properties:
+		            domain: localhost
+		            http:
+		              "/": 8000
+		        - type: manualscaler
+		          properties:
+		            replicas: 3
+		*/
```

Comment:
debug info?

Reply:
no, it helps to review the code because there's lot of content in test data yaml.